lib/datadog/kit/events.rb

```diff
@@ -7,41 +7,62 @@ module Datadog
   module Kit
     # Tracking events
     module Events
-      # Attach login event information to the trace
+      APPSEC_LOGIN_SUCCESS_EVENT = 'users.login.success'
+      APPSEC_LOGIN_FAILURE_EVENT = 'users.login.failure'
```

Comment:
On naming, it seems a bit weird to call these things `APPSEC_..._EVENT`s since we're inside `datadog/kit` and I thought the reason to have this inside kit rather than inside appsec is because this is intended to be generic.

(E.g. should we at least hide these constants?)

Reply:
Yeah I'm not too happy with the whole thing, but I don't feel like it belongs to `Datadog::AppSec` because of the level of abstraction: this is kind of sugar above Tracing (like `Datadog::Kit::Identity.set_user`) and glue between AppSec and Tracing.

Reply:
I agree with @ivoanjo here: I think this concrete implementation should be isolated to `Security`.

Is `Kit` supposed to be product aware? If so, when and why would someone use `Kit`? I think I need clarity about the purpose of this namespace.

Reply:
> should we at least hide these constants

What would be the ddtrace-idiomatic way to "hide" constants?

Reply:
I usually use `private_constant` for that:

https://github.com/DataDog/dd-trace-rb/blob/1dac1c1281e3fc3672037a2b18957f54491060dd/lib/datadog/profiling.rb#L11

Reply:
Can we use `private_constant`? 

I grep the codebase and it seems we use it in some places

Reply:
One interesting thing we could consider is having private modules within the scope ok `Datadog::Kit`

Example: 
```ruby
module Datadog
  module Kit
    # Tracking events
    module EventSDK
      class << self
        def track(namespace, event, trace, **rest)
          puts namespace
          puts event
          puts trace
          puts rest
        end
      end
    end

    private_constant :EventSDK

    module Events

      module Appsec
        LOGIN_SUCCESS_EVENT = 'users.login.success'
        LOGIN_FAILURE_EVENT = 'users.login.failure'
      end

      def self.track_login_success(trace, user:, **others)
        EventSDK.track(:appsec, Appsec::LOGIN_SUCCESS_EVENT, nil, **others)
      end

      def self.track_login_failure(trace, user_id:, user_exists:, **others)
        EventSDK.track(:appsec, Appsec::LOGIN_FAILURE_EVENT, nil, **others)
      end

      def self.track(namespace, event_name, trace, user:, **others)
        EventSDK.track(namespace, event_name, nil, **others)
      end
    end
  end
end


Datadog::Kit::Events.track_login_success(nil, user: "me")

DataDog::Kit::EventSDK.track(:appsec, Appsec::LOGIN_FAILURE_EVENT, nil)
# test.rb:42:in `<main>': uninitialized constant DataDog (NameError)

# DataDog::Kit::EventSDK.track(:appsec, Appsec::LOGIN_FAILURE_EVENT, nil)
# ^^^^^^^^^^
# Did you mean?  Datadog
```